source/backtrace/backtrace.d

```diff
@@ -196,59 +232,96 @@ private void printPrettyTrace(const(void*[]) bt, File output, PrintOptions optio
 
       lines.drop(startingLine);
       auto lineNumber = startingLine + 1;
-      output.writeln();
+      output.put(insertNewlines?"\n":"");
       foreach (line; lines.take(endingLine - startingLine)) {
-        output.writeln(forecolor(t.line == lineNumber ? Color.yellow : Color.cyan), t.line == lineNumber ? ">" : " ", "(", lineNumber , ") ", forecolor(t.line == lineNumber ? Color.yellow : Color.blue), line);
+        auto s2 = appender!string();
+        formattedWrite(
+          s2,
+          "%s%s(%d)%s%s",
+          forecolor(t.line == lineNumber ? Color.yellow : Color.cyan),
+          t.line == lineNumber ? ">" : " ",
+          lineNumber,
+          forecolor(t.line == lineNumber ? Color.yellow : Color.blue),
+          line
+        );
+        output.put(s2.data);
+        if (insertNewlines) output.put("\n");
         lineNumber++;
       }
-      output.writeln(reset());
-
+      output.put(reset());
+      if (insertNewlines) output.put("\n");
     }
 
     if (options.stopAtDMain && symbol == "_Dmain") break;
   }
 }
 
+// TODO. I imagine something like this is available somewhere in Phobos.
```

Comment:
If I remember correctly, you can pass the delegate to `formattedWrite` without the need for these.